lib/Optimizer/Dialect/Quake/QuakeOps.cpp

```diff
@@ -54,6 +54,54 @@ LogicalResult quake::verifyWireArityAndCoarity(Operation *op) {
   return op->emitOpError("arity does not equal coarity of wires");
 }
 
+bool quake::isSupportedMappingOperation(Operation *op) {
+  return isa<OperatorInterface, MeasurementInterface, SinkOp>(op);
+}
+
+mlir::ValueRange quake::getWiresFromRange(mlir::ValueRange range) {
+  mlir::ValueRange retWires = range;
+
+  // Skip over non-wires at the beginning
+  int numNonWires = 0;
+  for (auto operand : retWires) {
+    if (!isa<RefType, VeqType, WireType>(operand.getType()))
+      numNonWires++;
```

Comment:
Is this correct? Ref and Veq are not wires.

Reply:
> Is this correct? Ref and Veq are not wires.

Per our discussion, I renamed many of these "wires" to "quantum" in 51b626dd since they are referring to quantum types.